lib/circuit.js

```diff
@@ -222,6 +232,17 @@ class CircuitBreaker extends EventEmitter {
         handleError(error, this, timeout, args, resolve, reject);
       }
     });
+    if (this.options.cache) {
```

Comment:
Yeah, my comment is in the wrong line of code. Essentially, I was saying, in the ctor, we could check the type of object. If boolean, use our WeakMap - if Object, Map or WeakMap, use what's provided. In either case, set a flag that the check here would use. Make sense?